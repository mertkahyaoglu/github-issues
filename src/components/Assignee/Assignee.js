import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Avatar } from 'styles';

const Assignee = ({ assignee }) => (
  <StyledAssignee target="_blank" href={assignee.html_url}>
    <Avatar src={assignee.avatar_url} alt={assignee.login} />
  </StyledAssignee>
);

Assignee.propTypes = {
  assignee: PropTypes.shape({
    avatar_url: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired,
    login: PropTypes.string.isRequired,
  }).isRequired,
};

export default Assignee;

const StyledAssignee = styled.a`
  margin-right: 4px;
`;
