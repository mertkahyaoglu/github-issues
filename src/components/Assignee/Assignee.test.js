import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Assignee from './Assignee';
import TestProviders from 'utils/TestProviders';

describe('component/Assignee', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <TestProviders>
        <Assignee assignee={assignee} />
      </TestProviders>,
      div
    );
  });

  it('renders markup correctly', () => {
    const tree = renderer
      .create(
        <TestProviders>
          <Assignee assignee={assignee} />
        </TestProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const assignee = {
  login: 'mertkahyaoglu',
  id: 7414026,
  avatar_url: 'https://avatars0.githubusercontent.com/u/7414026?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/mertkahyaoglu',
  html_url: 'https://github.com/mertkahyaoglu',
  followers_url: 'https://api.github.com/users/mertkahyaoglu/followers',
  following_url:
    'https://api.github.com/users/mertkahyaoglu/following{/other_user}',
  gists_url: 'https://api.github.com/users/mertkahyaoglu/gists{/gist_id}',
  starred_url:
    'https://api.github.com/users/mertkahyaoglu/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/mertkahyaoglu/subscriptions',
  organizations_url: 'https://api.github.com/users/mertkahyaoglu/orgs',
  repos_url: 'https://api.github.com/users/mertkahyaoglu/repos',
  events_url: 'https://api.github.com/users/mertkahyaoglu/events{/privacy}',
  received_events_url:
    'https://api.github.com/users/mertkahyaoglu/received_events',
  type: 'User',
  site_admin: false,
};
