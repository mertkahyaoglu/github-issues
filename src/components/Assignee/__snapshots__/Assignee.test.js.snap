// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Assignee renders markup correctly 1`] = `
<a
  className="sc-bwzfXH kdDOvI"
  href="https://github.com/mertkahyaoglu"
  target="_blank"
>
  <img
    alt="mertkahyaoglu"
    className="sc-bdVaJa jrKHmD"
    src="https://avatars0.githubusercontent.com/u/7414026?v=4"
  />
</a>
`;
